tests/components/wmspro/test_light.py

```diff
@@ -0,0 +1,205 @@
+"""Test the wmspro cover support."""
+
+from unittest.mock import AsyncMock, patch
+
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.light import ATTR_BRIGHTNESS
+from homeassistant.components.wmspro.const import DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.setup import async_setup_component
+
+from . import setup_config_entry
+
+from tests.common import MockConfigEntry
+
+
+async def test_light_device(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_dimmer: AsyncMock,
+    device_registry: dr.DeviceRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test that a light device is created correctly."""
+    assert await setup_config_entry(hass, mock_config_entry)
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_dimmer.mock_calls) == 2
+
+    device_entry = device_registry.async_get_device(identifiers={(DOMAIN, "97358")})
+    assert device_entry is not None
+    assert device_entry == snapshot
+
+
+async def test_light_update(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_dimmer: AsyncMock,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test that a light entity is created and updated correctly."""
+    assert await setup_config_entry(hass, mock_config_entry)
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_dimmer.mock_calls) == 2
+
+    entity = hass.states.get("light.licht")
+    assert entity is not None
+    assert entity == snapshot
+
+    await async_setup_component(hass, "homeassistant", {})
+    await hass.services.async_call(
+        "homeassistant",
+        "update_entity",
+        {ATTR_ENTITY_ID: entity.entity_id},
+        blocking=True,
+    )
```

Comment:
Please don't use the update entity service for this, rather let the entity naturally update, for example with either freezing and skipping time or by calling a callback function fetched from a mock

Reply:
Time skipping it is, thanks for the hint!